ambry-cloud/src/main/java/com.github.ambry.cloud/CloudStorageManager.java

```diff
@@ -109,20 +108,22 @@ public boolean removeBlobStore(PartitionId id) {
   @Override
   public ServerErrorCode checkLocalPartitionStatus(PartitionId partition, ReplicaId localReplica) {
     /*
-     * Vcr cluster map determines if a vcr node can handle requests for a particular replica.
-     * So either a vcr node will never get a partition assigned to it, or has a partition removed
-     * from it, in both of those cases we return Partition_Unknown. In case the partition was assigned,
-     * but is in stopped state currently, we return Replica_Unavailable.
-     * The actual disk and data for vcr node resides on azure. So a partition will likely never be
-     * actually unavailable. The Partition_Unknown and Replica_Unavailable error codes only determine if
-     * the partition is unknown or replica is unavailable with respect to this vcr node.
+     * Vcr nodes serve requests for all partitions from cloud blob store. So assuming that cloud is always available,
+     * the local partition status for a vcr node should always be available.
      */
-    if (partitionToStore.get(partition) == null) {
-      return ServerErrorCode.Partition_Unknown;
-    }
-    if (!partitionToStore.get(partition).isStarted()) {
-      return ServerErrorCode.Replica_Unavailable;
-    }
     return ServerErrorCode.No_Error;
   }
+
+  private void createAndStartBlobStoreIfAbsent(PartitionId partitionId) {
+    partitionToStore.computeIfAbsent(partitionId, value -> {
+      CloudBlobStore store = null;
+      try {
+        store = new CloudBlobStore(properties, partitionId, cloudDestination, clusterMap, vcrMetrics);
+        store.start();
```

Comment:
We should avoid side effects like starting a store inside a computeIfAbsent lambda.

Reply:
fixed.

Reply:
Add a log after `CloudBlobStore  created successfully?